src/mscorlib/src/System/Span.cs

```diff
@@ -625,5 +636,266 @@ internal static unsafe void CopyTo<T>(ref T destination, ref T source, int eleme
                 }
             }
         }
+
+        internal static unsafe void ClearWithoutReferences(ref byte b, nuint byteLength)
+        {
+            if (byteLength == 0)
+                return;
+
+            fixed (byte* bytePointer = &b)
+            {
+                // Note: It's important that this switch handles lengths at least up to 22.
+                // See notes below near the main loop for why.
+
+                // The switch will be very fast since it can be implemented using a jump
+                // table in assembly. See http://stackoverflow.com/a/449297/4077294 for more info.
+
+                switch (byteLength)
```

Comment:
My general understanding is that memset also does tricks like these. Do you have data showing that it's not faster to just unconditionally call RhZeroMemory?